pyodide-build/pyodide_build/out_of_tree/pypi.py

```diff
@@ -0,0 +1,377 @@
+import json
+import os
+import shutil
+import sys
+import tempfile
+from collections.abc import Generator
+from contextlib import contextmanager
+from email.message import EmailMessage
+from email.parser import BytesParser
+from io import BytesIO
+from operator import attrgetter
+from pathlib import Path
+from platform import python_version
+from typing import Any, BinaryIO, cast
+from urllib.parse import urlparse
+from zipfile import ZipFile
+
+import requests
+from packaging.requirements import Requirement
+from packaging.utils import canonicalize_name
+from packaging.version import Version
+from resolvelib import BaseReporter, Resolver
+from resolvelib.providers import AbstractProvider
+from unearth.evaluator import TargetPython
+from unearth.finder import PackageFinder
+
+from .. import common
+from . import build
+
+
+@contextmanager
+def stream_redirected(to=os.devnull, stream=None):
```

Comment:
Maybe you could use contextlib.redirect_stderr and redirect_stdout to avoid re-implementing it.

Reply:
contextlib doesn't redirect subprocess output - that's why this one is needed. Otherwise one would need to delve into anywhere subprocess is called (i.e. deep inside build) to redirect that.

Reply:
Could you please write a comment about it for the people who read this code in the future?